parso/python/errors.py

```diff
@@ -961,12 +975,26 @@ def _check_assignment(self, node, is_deletion=False, is_namedexpr=False):
               or '_test' in type_
               or type_ in ('term', 'factor')):
             error = 'operator'
+        elif type_ == 'name':
+            if node.value == '__debug__':  # should be a keyword for all versions
+                if (2, 7) < self._normalizer.version < (3, 8):
+                    error = 'keyword'
+                else:
+                    error = str(node.value)
+                not_debug = False
+                is_keyword = True
 
         if error is not None:
-            if is_namedexpr:
+            if is_namedexpr and not_debug:
+                # c.f. CPython bpo-39176, should be changed in next release
+                # message = 'cannot use assignment expressions with %s' % error
                 message = 'cannot use named assignment with %s' % error
             else:
-                cannot = "can't" if self._normalizer.version < (3, 8) else "cannot"
+                _version = self._normalizer.version
+                if _version[0] < 3:  # Python 2.*
+                    cannot = ("can not" if _version < (2, 7) else "cannot") if is_keyword else "can't"
```

Comment:
IMO ignore Python2.6 here. I will just drop support for it. Nobody uses that anymore.

Reply:
Also I wouldn't add the keyword variable. The messages don't need to match exactly as long as they are at least there. If it tests cause any trouble, add an additional case to  `_get_actual_exception` in `test_python_errors.py`.

Reply:
Okay, I removed all these crazy cases and just patched the "actual" exceptions.

Reply:
Ah... seems you just dropped Python 2... lol

Reply:
> Ah... seems you just dropped Python 2... lol

All because of you, hahaha :D